### PR TITLE
fix: fill missing fees in tx object received from dApp

### LINKED ISSUE #16528: [Wallet connect] Fill missing fees in tx object received from dApp
# Bug Report

## Description

Some dApps will propose an incomplete transaction object where fees are missing. We'll need to compute the fees fill the tx object on our end.

This can be reproduced when using the revoke token approvals dApp on etherscan or optimistic.etherscan. In the logs we'll se an error when parsing the fees and the sign message will show null fees in the tx object description:

Image

```
cannot convert fees info to hex: DBG 2024-10-15 15:44:03.065Z NewßE_callPrivateRPC tid-63326574 file-helpers.nim:42 msg-"input(1, 7) Error: { expected"
```

## Steps to reproduce

1. Make sure you sign token approvals first. Can be done with Uniswap
2. Go to https://optimistic.etherscan.io/tokenapprovalchecker?search=0x4f014b9143c52a9f07b7edf9fd79ca2176532140 and connect the dApp to an account with token approvals
3. Revoke the token approval
#### Expected behavior
The tx object has the fees and there no error in the log related to tx processing
#### Actual behavior
The tx object does not have the fees and there is no error in the log related to tx processing

### PR DESCRIPTION
**Pull request by saledjenic**
- the following params were removed from the tx preview (cause they are dynamic and recalculate with base fee change):
  - `maxFeePerGas`
  - `maxPriorityFeePerGas`
  - `gasPrice`
- fees and estimated time are recalculated periodically

Fixes #16528

https://github.com/user-attachments/assets/9282d927-320b-4d84-9704-542faa5cc235

### CODE REVIEW

**Comment by saledjenic** [comment]
Animated fees change added


https://github.com/user-attachments/assets/dc8e16a8-fedd-4c79-a84a-9edd605b8efb

**Comment by saledjenic** [comment]
@alexjba good point, do you want me to do this change in this PR?

**Comment by caybro** [comment]
> @alexjba good point, do you want me to do this change in this PR?

Yes please, and you can do that from QML with sth like:
```qml
enabled: Qt.application.state === Qt.ApplicationActive
```

**Comment by alexjba** [comment]
> @alexjba good point, do you want me to do this change in this PR?

Would be nice to have it here.

**Comment by saledjenic** [comment]
Updated. Thanks.

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 1]
```diff
@@ -158,6 +164,108 @@ SQUtils.QObject {
...
+                                     })
+
+            for (let i = 0; i < feesSubscription.chainIds.length; i++) {
+                if (feesSubscription.chainIds == request.chainId) {
+                    return
+                }
+            }
+
+            feesSubscription.chainIds.push(request.chainId)
+            feesSubscription.restart()
```
[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:234]: I did test it and it worked properly.

[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:234]: That one is called when the user closes the popup, what I was testing:
- open the popup -> fees were updating
- put the app in the background -> fees were not updating
- bring the app back to the front -> fees were updating
- reject that popup and started a new request from the dapp, repeating the process above, and everything looked good

**Review by alaibe — APPROVED** [review]

**Pull request merged**